expr/sort.go

```diff
@@ -114,13 +122,19 @@ func sortByBraces(metrics []*types.MetricData, part int, pattern string) {
 	}
 }
 
+// compile regex for sorting
+// it splits string by dots, but ignoring enclosed {} and []
+// it fixes bug when number of dots in pattern and metric can differ
+// causing out of index error in getParts() above
+var reSplitByDotsIgnoreBraces = regexp.MustCompile(`\s*(\{[^}]*\}|\[[^]]*\]|[^.]+)`)
```

Comment:
It is not a good idea to use a global variable unless there are good reasons for it. Sorry, but I don't see a reason to use a global variable here. Please consider making this variable local to where it's used.

I guess, it is global to prevent repeated re-compilation. I don't think this is a sufficient reason to have it as a global.

On top of that, since this is not a simple constant declaration, this will include unexpected and implicit code execution during initialization. This can create many problems.

Reply:
@grzkv : Having regex in global variables is common pattern, common enough to have it [excluded](https://github.com/leighmcculloch/gochecknoglobals/issues/6) in [gochecknoglobals](https://github.com/leighmcculloch/gochecknoglobals) even.
But in this case it's probably fine, it will be called only once per target, so, I'll make it local.

Reply:
I implemented split function, so, probably not relevant anymore.